Automattic/wp-calypso
# Marketing Connections: Fix the tour position

#### Changes proposed in this Pull Request

The tour displayed on the marketing connections page was being
incorrectly positioned at the top left. This was due to the target
element not existing until the connection statuses had been received for
the API

The guided tour API has a `wait` property, which is a function that will
prevent the guide from appearing until it returns. This change copies a
pattern from another Jetpack guide, which polls for the existence of the
element. I'm not convinced that's a great solution. I think it would be
better to have the guide itself wait for the existence of the target
element rather than default to the top left of the screen. This could
have other consequences, and so this fix seems reasonable for now, until
this can be looked at more closely.

The `meta` settings also made this tour appear by simply visiting the
`/marketing/connections` route on a desktop. This wasn't intended and
should have be left to be triggered by the `tour` parameter added from
the Customer Home card.

Lastly, on my test site with a premium plan, a prompt to upgrade
disappeared, but the tour didn't reposition until I scrolled or resized
the window. Setting the `keepRepositioning` prop helps with this, but
that only polled once every three seconds. I've reduced that to 700ms,
so there's less jank. Opinions on this change are welcome!

Props to @allilevine for debugging this with me.

#### Testing instructions

* Go to `/marketing/connections/?tour=marketingConnectionsTour`
* Without this change, the tour should display incorrectly at the top left of the screen
* With this change, the tour should appear once the connection statuses have loaded and as the screen changes it should adjust its position.

Fixes #44194
